src/bio/bio_context.c

```diff
@@ -277,10 +277,38 @@ bio_bs_hold(struct bio_blobstore *bbs)
 	return rc;
 }
 
+struct bio_media_error_msg {
+	struct bio_blobstore	*bmem_bs;
+	int			 bmem_rc;
+};
+
+/*
+ * MEDIA ERROR event.
+ * Store BIO I/O error in in-memory device state. Called from device owner
+ * xstream only.
+ */
+static void
+bio_media_error(void *msg_arg)
+{
+	struct bio_media_error_msg	*bmem = msg_arg;
+	struct bio_device_health_state  *hs;
+
+	hs = &bmem->bmem_bs->bb_dev_health.bhm_health_state;
+	hs->bhs_bio_err = bmem->bmem_rc;
```

Comment:
For blob I/O and unmap, I suppose SPDK returns EIO only, so the error code is probably not very useful here. Instead, we'd log error counters for read, write & unmap.